rpc/net.go

```diff
@@ -73,6 +88,71 @@ func (s *NetService) Peers(_ *http.Request, _ *interface{}, resp *rpctypes.Peers
 	return nil
 }
 
+// Pairs returns all currently available pairs from offers of all peers
+func (s *NetService) Pairs(_ *http.Request, req *rpctypes.PairsRequest, resp *rpctypes.PairsResponse) error {
+	if s.isBootnode {
+		return errUnsupportedForBootnode
+	}
+
+	peerIDs, err := s.discover(req)
+	if err != nil {
+		return err
+	}
+
+	addrs := make(map[ethcommon.Address]int)
+	pairs := make([]*types.Pair, 0, 100)
+
+	addrs[ethcommon.Address{}] = 0 // ETH/XMR pair is always first
+	pairs = append(pairs, types.NewPair(types.EthAsset{}))
+	pairs[0].Verified = true
+	pairs[0].Token = coins.ERC20TokenInfo{
+		NumDecimals: 18,
+		Name:        "Ether",
+		Symbol:      "ETH",
+	}
+
+	for _, p := range peerIDs {
+		msg, err := s.net.Query(p)
+		if err != nil {
+			log.Debugf("Failed to query peer ID %s", p)
+			continue
+		}
+		if len(msg.Offers) > 0 {
+			for _, o := range msg.Offers {
+				address := o.EthAsset.Address()
+				index, exists := addrs[address]
+				pair := types.NewPair(o.EthAsset)
+				if !exists {
+					addrs[address] = index
```

Comment:
if the value does not exist in the map, isn't `index` always 0? this seems off, can you simplify this logic to just have one map `pairs := make(map[ethcommon.Address]*types.Pair)` where the key is the ETH-asset?